### PR TITLE
Adds general sidenav component

### PR DESCRIPTION
**Pull request by tesk9**
There are definitely accessibility improvements and API improvements left to make here, but this is a start!

Step 1 of https://trello.com/c/SPSSTOZq/487-style-guide-consolidate-and-add-sidebar-nav See https://noredink.slack.com/archives/C9MEFL3KL/p1638548734064000 for some spec clarification.

<details><summary>Old screenshots</summary>

Screen Shot 2021-12-03 at 12 04 37 PM

---

Screen Shot 2021-12-03 at 12 11 40 PM

</details>

cc @NoRedInk/design

### CODE REVIEW

**Comment by bendansby** [comment]
Cool! Ready for style notes yet or nah?

**Comment by tesk9** [comment]
Definitely would love style notes!!

**Comment by bendansby** [comment]
Kinda looks like the styles of the first and second screenshots need to merge. The first has the right container padding, while the second has the right distance between items.

**Comment by tesk9** [comment]
How do these look?

Screen Shot 2021-12-03 at 1 09 42 PM
Screen Shot 2021-12-03 at 1 09 48 PM

**Comment by bendansby** [comment]
Perfect!

File: src/ClickableAttributes.elm [hunk 1]
```diff
@@ -102,12 +112,24 @@ toButtonAttributes clickableAttributes =
 
 
 {-| -}
-toLinkAttributes : ClickableAttributes msg -> ( String, List (Attribute msg) )
-toLinkAttributes clickableAttributes =
+toLinkAttributes : (route -> String) -> ClickableAttributes route msg -> ( String, List (Attribute msg) )
+toLinkAttributes routeToString clickableAttributes =
```
Comment (juanedi) [inline src/ClickableAttributes.elm:116]: 🎨  It feels like there are some implicit rules here about what the valid combinations of `url`/`urlString` and `linkType` are, and I wonder if modelling those would simplify this function and make this easier to understand overall.

I'm thinking something like this:

```elm
type alias ClickableAttributes route msg =
    { linkType : Link route
    , onClick : Maybe msg
    }


type Link route
    = Default
    | WithTracking route
    | SinglePageApp route
    | WithMethod String route
    | External String
    | ExternalWithTracking String
```

It seems like:
  - `Default` is the case in which there isn't any target `url`/`urlString`
  - We only use `urlString` with external links
  - For all other cases we use `url`.

If this made sense, we could implement this with a single case, grabbing the target link from the `Link`'s constructor params.

**Review by juanedi — COMMENTED** [review]
This looks awesome! 😻 😻 😻 

Left just one suggestion with a potential refactor in `ClickableAttributes`. If that doesn't make sense for any reason feel free to merge!

File: src/ClickableAttributes.elm (hunk 1 above)
Comment (tesk9) [reply src/ClickableAttributes.elm:116]: Thanks for the suggestion here! I'm going to link this in an issue, and come back to it when I've got a little more bandwidth. 🙏

**Pull request merged**